### PR TITLE
Add recipe for libtiledb-sql-py

### PR DESCRIPTION
**Pull request by Shelnutt2**
We use a `git_url` instead of a tarball due to the use of `setuptools_scm` in the python package.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/libtiledb-sql-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{% set name = "libtiledb-sql-py" %}
+{% set version = "2.0.0" %}
+{% set sha256 = "c0228d64bdf9eeabdac5284c760bf4899892a170f769688a09285bcbd46963d1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/TileDB-Inc/TileDB-SQL-Py.git
```
Comment (scopatz) [inline recipes/libtiledb-sql-py/meta.yaml]: Please build from a source tarball: https://conda-forge.org/docs/maintainer/adding_pkgs.html#tarballs-no-repos

File: recipes/libtiledb-sql-py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    Upstream source: https://github.com/PyMySQL/mysqlclient-python
+
+  doc_url: https://docs.tiledb.com/developer/api-usage/embedded-sql
+  dev_url: https://github.com/TileDB-Inc/TileDB-SQL-Py
+
+extra:
+  recipe-maintainers:
+    - shelnutt2
+    - ihnorton
```
Comment (scopatz) [inline recipes/libtiledb-sql-py/meta.yaml]: @ihnorton are you OK with being a maintainer?

Comment (ihnorton) [reply recipes/libtiledb-sql-py/meta.yaml]: yes

File: recipes/libtiledb-sql-py/meta.yaml (hunk 1 above)
Comment (Shelnutt2) [reply recipes/libtiledb-sql-py/meta.yaml]: I've reworked the package to use `setuptools_scm_git_archive`, it no longer needs the git repo and now uses a github archive.

File: recipes/libtiledb-sql-py/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  host:
+    - python
+    - wheel
+    - setuptools
+    - setuptools_scm
+    - setuptools_scm_git_archive
+    - libtiledb-sql
+  run:
+    - python
+    - setuptools
```
Comment (scopatz) [inline recipes/libtiledb-sql-py/meta.yaml]: Do you really need setuptools at runtime?

Comment (Shelnutt2) [reply recipes/libtiledb-sql-py/meta.yaml]: No, this was a mistake, the recipe has been updated.

**Pull request merged**